src/pudl_archiver/orchestrator.py

```diff
@@ -402,12 +357,50 @@ async def _update_datapackage(self, resources: dict[str, ResourceInfo]):
             bytes(datapackage.json(by_alias=True, indent=4), encoding="utf-8")
         )
 
-        await self._apply_change(
-            _DepositionChange(
-                action_type=_DepositionAction.CREATE,
-                name="datapackage.json",
-                resource=datapackage_json,
+        if old_datapackage is None:
+            await self._apply_change(
+                _DepositionChange(
+                    action_type=_DepositionAction.CREATE,
+                    name="datapackage.json",
+                    resource=datapackage_json,
+                )
             )
-        )
+        else:
+            if self._datapackage_worth_changing(old_datapackage, datapackage):
+                await self._apply_change(
+                    _DepositionChange(
+                        action_type=_DepositionAction.UPDATE,
+                        name="datapackage.json",
+                        resource=datapackage_json,
+                    )
+                )
 
         return datapackage, old_datapackage
+
+    def _datapackage_worth_changing(
+        self, old_datapackage: DataPackage, new_datapackage: DataPackage
+    ) -> bool:
+        # ignore differences in created/version
+        # ignore differences resource paths if it's just some ID number changing...
+        for field in new_datapackage.dict():
+            if field in {"created", "version"}:
+                continue
+            if field == "resources":
+                for r in old_datapackage.resources + new_datapackage.resources:
+                    r.path = re.sub(r"/\d+/", "/ID_NUMBER/", str(r.path))
```

Comment:
I'm not happy about this heuristic for "some ID number has changed, please ignore," but I'm also not sure what a better version of this would look like.

Reply:
Well - the URLs are sort of important. If we update `datapackage.json` to use the "canonical" URL of `/records/<record_id>/files/<file_name>` vs. whatever the hell `self.links.download` gives us (`/api/records/<record_id>(/draft)?/files/<file_name>(/content)?`), then we *do* want that to register as a change in the datapackage. But if we see the format stay the same & the record ID increment, then we don't want that to register as a change.

Reply:
Overriding `__eq__` seemed like a too-general place to do this weird and specific logic to me - there are lots of ways to define equality of `Resource`s and I wouldn't presume to promote this one to default.